hashmap.h

```diff
@@ -288,6 +300,34 @@ int hashmap_iterate(const struct hashmap_s *const m,
   return 0;
 }
 
+int hashmap_iterate_pairs(struct hashmap_s *const m,
+                    int (*f)(void *const, const char *, unsigned, void *const), void *const context) {
+  unsigned int i;
+  hashmap_element_s *p;
+  int r;
+
+  /* Linear probing */
+  for (i = 0; i < m->table_size; i++)
+    p=&m->data[i];
+    if (p->in_use) {
+      r=f(context, p->key, p->key_len, p->data);
```

Comment:
why not:

```
switch( f(context, &m->data[i] ) ) {
  case -1: ...
  case 0: ...
  default: ...
}
```
i pretty much dislike passing `key`, `key_len` and `data` to the callback. passing just the node is the most generic from and could even replace the existing iterator. of course it would make the internal node struct visible.

Reply:
If the in_use field is set to 0 (false) it would introduce a logic error to not check it.  I will use a switch statement though.